### PR TITLE
Fix Coverity defects

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

**Test plan**

CI is green

### CODE REVIEW

**Comment by XVilka** [comment]
@ret2libc a couple more issues for dist scripts:
On "Build OSX package":
```
Run ./dist/osx/build_osx_package.sh
Traceback (most recent call last):
  File "./sys/version.py", line 15, in <module>
    with open(meson_file, "r", encoding="utf8") as f:
TypeError: 'encoding' is an invalid keyword argument for this function
Error: Process completed with exit code 1.
```

File: librz/core/project.c [hunk 1]
```diff
@@ -44,19 +44,20 @@ RZ_API RzProjectErr rz_project_save_file(RzCore *core, const char *file, bool co
 
 	if (compress) {
 		int mkstemp_fd = rz_file_mkstemp("svprj", &tmp_file);
-		close(mkstemp_fd);
```
Comment (ret2libc) [inline librz/core/project.c:47]: why this?

Comment (XVilka) [reply librz/core/project.c:47]: In case of `mkstem_fd` is -1 `close()` is unadvised AFAIK.

Comment (ret2libc) [reply librz/core/project.c:47]: right..

**Review by ret2libc — APPROVED** [review]

**Pull request merged**